BH/Modules/Item/Item.cpp

```diff
@@ -157,13 +163,372 @@ void GetItemStats()
+	{
+		flags |= ITEM_GROUP_DRUID_PELT;
+	}
+	return flags;
+}
+
+std::string GetTxtItemCode(ItemsTxt* pItemsTxt)
+{
+	char code[5] = {};
+	code[0] = pItemsTxt->szCode[0];
+	code[1] = pItemsTxt->szCode[1] != ' ' ? pItemsTxt->szCode[1] : 0;
+	code[2] = pItemsTxt->szCode[2] != ' ' ? pItemsTxt->szCode[2] : 0;
+	code[3] = pItemsTxt->szCode[3] != ' ' ? pItemsTxt->szCode[3] : 0;
+	code[4] = 0;
+
+	return code;
+}
+
+void GetWeaponAttributes()
+{
+	D2ItemDataTbl* pItemDataTables = D2COMMON_10535_DATATBLS_GetItemDataTables();
+
+	for (auto d = 0; d < pItemDataTables->nWeaponsTxtRecordCount; d++)
+	{
+		ItemsTxt* pWeapons = &pItemDataTables->pWeapons[d];
+		BYTE stackable = pWeapons->bstackable > 0 ? pWeapons->bstackable : 0;
+		BYTE useable = pWeapons->buseable > 0 ? pWeapons->buseable : 0;
+		BYTE throwable = throwableMap[pWeapons->nType] > 0 ? throwableMap[pWeapons->nType] : 0;
+		unsigned int flags = ITEM_GROUP_ALLWEAPON, flags2 = 0;
+
+		std::set<WORD> ancestorTypes;
+		FindAncestorTypes(pWeapons->nType, ancestorTypes, parentMap1, parentMap2);
+
+		if (pWeapons->dwcode == pWeapons->dwultracode)
+		{
+			flags |= ITEM_GROUP_ELITE;
+		}
+		else if (pWeapons->dwcode == pWeapons->dwubercode)
+		{
+			flags |= ITEM_GROUP_EXCEPTIONAL;
+		}
+		else
+		{
+			flags |= ITEM_GROUP_NORMAL;
+		}
+
+		if (ancestorTypes.find(ITEM_TYPE_CLUB) != ancestorTypes.end() ||
+			ancestorTypes.find(ITEM_TYPE_HAMMER) != ancestorTypes.end() ||
+			ancestorTypes.find(ITEM_TYPE_MACE) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_MACE;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_WAND) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_WAND;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_STAFF) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_STAFF;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_BOW) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_BOW;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_AXE) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_AXE;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_SCEPTER) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_SCEPTER;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_SWORD) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_SWORD;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_KNIFE) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_DAGGER;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_SPEAR) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_SPEAR;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_POLEARM) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_POLEARM;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_CROSSBOW) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_CROSSBOW;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_JAVELIN) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_JAVELIN;  // TODO: this isn't being used. Fixme
+		}
+		if (ancestorTypes.find(ITEM_TYPE_THROWN) != ancestorTypes.end())
+		{
+			flags |= ITEM_GROUP_THROWING;
+		}
+
+		ItemAttributes* attrs = new ItemAttributes();
```

Comment:
Just marking these as potential issues, needs the same treatment as the other comments